File: io/lucamienert/basic/ast/ImmediateExpression.java
package io.lucamienert.basic.ast;

import io.lucamienert.basic.instruction.Instruction;
import io.lucamienert.basic.instruction.InstructionSequence;
import io.lucamienert.basic.instruction.OpCode;

public class ImmediateExpression extends Expression {

    private final int value;

    public ImmediateExpression(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    @Override
    public void compile(InstructionSequence seq) {
        seq.append(new Instruction(OpCode.PUSHI, value));
    }
    
}

File: src/BasicCompiler/Ast/ImmediateExpression.cs
using BasicCompiler.Instructions;

namespace BasicCompiler.Ast;

public class ImmediateExpression : Expression
{
    public int Value { get; set; }

    public ImmediateExpression(int value)
        => Value = value;

    public override void Compile(InstructionSequence instructionSequence)
    {
        instructionSequence.Append(new Instruction(OpCode.PUSHI, Value));
    }
}
